qt/scientific_interfaces/Indirect/JumpFitModel.cpp

```diff
@@ -281,28 +280,47 @@ std::string JumpFitModel::getFitParameterName(TableDatasetIndex dataIndex,
 }
 
 void JumpFitModel::setActiveWidth(std::size_t widthIndex,
-                                  TableDatasetIndex dataIndex) {
+                                  TableDatasetIndex dataIndex, bool single) {
   const auto parametersIt = findJumpFitParameters(dataIndex);
   if (parametersIt != m_jumpParameters.end() &&
       parametersIt->second.widthSpectra.size() > widthIndex) {
     const auto &widthSpectra = parametersIt->second.widthSpectra;
-
-    setSpectra(
-        createSpectra(std::vector<std::size_t>({widthSpectra[widthIndex]})),
-        dataIndex);
+    if (single == true) {
+      setSpectra(
+          createSpectra(std::vector<std::size_t>({widthSpectra[widthIndex]})),
+          dataIndex);
+    } else { // In multiple mode the spectra needs to be appending on the
+             // existing spectra list.
+      auto spectra_vec = std::vector<std::size_t>({widthSpectra[widthIndex]});
+      auto spectra = getSpectra(dataIndex);
+      for (size_t i = 0; i < spectra.size().value; i++) {
```

Comment:
Does the spectra object allow for each loops? My memory is that it doesn't buy if it does could use one here. I think the Spectra object going forward is one which could potentially be simplified. There are some edge cases to handle but a vector of int's would work for most cases.